settings/src/components/auto-tabbing-input.js

```diff
@@ -45,8 +45,21 @@ const AutoTabbingInput = ( props ) => {
 		}
 	}, [] );
 
+	const handlePaste = useCallback( ( event ) => {
+		const paste = event.clipboardData.getData( 'Text' ).replace( /[^0-9]/g, '' );
+
+		if ( inputs.length === paste.length ) {
```

Comment:
I wonder if it makes sense to show a message if the length doesn't match, eg. 'Please enter a six digit code'. I know there is text above instructing this, but without it I think it's hard to understand what's going wrong, say in the case where they have made a mistake copying.

We already have a notice component there which could display the message.

Reply:
It's also possible to paste letters in, we might need validation for that too, unless they're valid

![Screen Shot 2023-05-09 at 11 19 04 AM](https://user-images.githubusercontent.com/1017872/236957529-5f71933c-a1e3-48e6-b13a-e1d75afd81f7.jpg)

Reply:
> I wonder if it makes sense to show a message if the length doesn't match, eg. 'Please enter a six digit code'. 

I thought about whether it should do something in that scenario, but decided against it and just went for the whole "If you paste what we're expecting, process it, else let the rest of the UI code handle it"

> It's also possible to paste letters in, we might need validation for that too, unless they're valid

It's also possible to type letters in though, so I didn't really see that as much of an issue.. That might vary between browsers. I've only tested with Chrome.

Reply:
> It's also possible to type letters in though

Turns out this is validated on submit, so yes I think we can leave this alone as a rare edge case.

I have added a length validation error message though.